app/src/sharedTest/java/org/oppia/app/parser/FractionInteractionViewModelTest.kt

```diff
@@ -0,0 +1,155 @@
+package org.oppia.app.parser
+
+import android.content.Context
+import android.content.res.Configuration
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Test
```

Comment:
@BenHenning PTAL and let me know what other test cases you are expecting

Reply:
Did you port over the tests from Oppia web? This part is essential: unless we are sure the same logical test cases pass as web (e.g. https://github.com/oppia/oppia/blob/develop/core/templates/domain/objects/NumberWithUnitsObjectFactorySpec.ts#L43), we can't attempt to refactor to simplify the parser.